packages/maskbook/src/plugins/Wallet/services/chain.ts

```diff
@@ -31,10 +33,14 @@ export async function updateChainState() {
 
     // update chain state
     try {
-        ;[currentBlockNumberSettings.value, currentBalanceSettings.value] = await Promise.all([
-            getBlockNumber(),
-            currentAccountSettings.value ? getBalance(currentAccountSettings.value) : currentBalanceSettings.value,
-        ])
+        ;[currentBlockNumberSettings.value, currentBalanceSettings.value, currentMaskWalletBalanceSettings.value] =
+            await Promise.all([
+                getBlockNumber(),
+                currentAccountSettings.value ? getBalance(currentAccountSettings.value) : currentBalanceSettings.value,
```

Comment:
Please reduce request when `currentAccountSettings` is the same as the `currentAccountMaskWalletSettings`.